### PR TITLE
HDFS-14802. The feature of protect directories should be used in RenameOp

### PR DESCRIPTION
**Pull request by ferhui**
## NOTICE

Please create an issue in ASF JIRA before opening a pull request,
and you need to set the title of the pull request which starts with
the corresponding JIRA issue number. (e.g. HADOOP-XXXXX. Fix a typo in YYY.)
For more details, please see https://cwiki.apache.org/confluence/display/HADOOP/How+To+Contribute

### CODE REVIEW

**Comment by ferhui** [comment]
@steveloughran @ayushtkn PR for HDFS-14802

File: hadoop-common-project/hadoop-common/src/site/markdown/filesystem/filesystem.md [hunk 1]
```diff
@@ -1071,6 +1074,8 @@ entry-by-entry delete operation.
 This can break the expectations of client applications for O(1) atomic directory
 deletion, preventing the stores' use as drop-in replacements for HDFS.
 
+* [The feature of protected directories](#ProtectedDirectories) can be supported.
```
Comment (jojochuang) [inline hadoop-common-project/hadoop-common/src/site/markdown/filesystem/filesystem.md]: @steveloughran what's your take? I thought this feature only concerns HDFS. Feels awkward to me that we talk about this behavior in a file system specification. (To me this is an implementation detail)

**Comment by jojochuang** [comment]
@ferhui  could you remove the line 

> [The feature of protected directories](#ProtectedDirectories) can be supported.

Other than that I am +1

**Comment by ferhui** [comment]
@jojochuang Thanks for review.
Add commit to remove improper lines

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 38 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 71 | Maven dependency ordering for branch |
| +1 | mvninstall | 1093 | trunk passed |
| +1 | compile | 1012 | trunk passed |
| +1 | checkstyle | 162 | trunk passed |
| +1 | mvnsite | 169 | trunk passed |
| +1 | shadedclient | 1141 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 186 | trunk passed |
| 0 | spotbugs | 177 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 300 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 26 | Maven dependency ordering for patch |
| +1 | mvninstall | 115 | the patch passed |
| +1 | compile | 974 | the patch passed |
| +1 | javac | 974 | the patch passed |
| +1 | checkstyle | 163 | the patch passed |
| +1 | mvnsite | 167 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 1 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 762 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 194 | the patch passed |
| +1 | findbugs | 320 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 554 | hadoop-common in the patch passed. |
| -1 | unit | 5674 | hadoop-hdfs in the patch failed. |
| +1 | asflicense | 60 | The patch does not generate ASF License warnings. |
| | | 13169 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdfs.server.blockmanagement.TestUnderReplicatedBlocks |
|   | hadoop.hdfs.server.namenode.TestRedudantBlocks |
|   | hadoop.hdfs.tools.TestDFSZKFailoverController |
|   | hadoop.hdfs.tools.offlineEditsViewer.TestOfflineEditsViewer |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.4 Server=19.03.4 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1669/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1669 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient xml findbugs checkstyle |
| uname | Linux fd83a5cfb434 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / df6b316 |
| Default Java | 1.8.0_222 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1669/2/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1669/2/testReport/ |
| Max. process+thread count | 3876 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-hdfs-project/hadoop-hdfs U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1669/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 76 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 75 | Maven dependency ordering for branch |
| +1 | mvninstall | 1244 | trunk passed |
| +1 | compile | 1076 | trunk passed |
| +1 | checkstyle | 174 | trunk passed |
| +1 | mvnsite | 159 | trunk passed |
| +1 | shadedclient | 1241 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 181 | trunk passed |
| 0 | spotbugs | 211 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 359 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 25 | Maven dependency ordering for patch |
| +1 | mvninstall | 124 | the patch passed |
| +1 | compile | 1081 | the patch passed |
| +1 | javac | 1081 | the patch passed |
| +1 | checkstyle | 171 | the patch passed |
| +1 | mvnsite | 160 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 1 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 852 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 171 | the patch passed |
| +1 | findbugs | 358 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 603 | hadoop-common in the patch passed. |
| -1 | unit | 6733 | hadoop-hdfs in the patch failed. |
| +1 | asflicense | 57 | The patch does not generate ASF License warnings. |
| | | 14906 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdfs.server.datanode.TestBPOfferService |
|   | hadoop.hdfs.tools.offlineEditsViewer.TestOfflineEditsViewer |
|   | hadoop.hdfs.TestSafeMode |
|   | hadoop.hdfs.tools.TestDFSZKFailoverController |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.4 Server=19.03.4 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1669/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1669 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient xml findbugs checkstyle |
| uname | Linux 2efc8788bde6 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / df6b316 |
| Default Java | 1.8.0_222 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1669/3/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1669/3/testReport/ |
| Max. process+thread count | 2975 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-hdfs-project/hadoop-hdfs U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1669/3/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by ferhui** [comment]
Failed tests are unreleated. They passed locally

**Comment by jojochuang** [comment]
+1 merging this

**Pull request merged**